File: benchmark/cases/go-hardcoded-credentials/source.go
// Case: Hardcoded database credentials in a Go service.
package db

const (
	// BUG: production database credentials are committed in plaintext.
	dsnUser = "billing_admin"
	dsnPass = "supersecret-prod-2024"
	dsnHost = "10.0.0.5:5432"
	dsnName = "billing"
)

func DSN() string {
	return "postgres://" + dsnUser + ":" + dsnPass + "@" + dsnHost + "/" + dsnName
}

File: benchmark/cases/ts-hardcoded-credentials/source.ts
// Case: Hardcoded database credentials in a TypeScript service config.
export const dbConfig = {
  host: 'db.prod.internal',
  port: 5432,
  user: 'admin',
  // BUG: the production database password is committed in plaintext.
  password: 'P@ssw0rd-prod-2024!',
  database: 'orders',
};

export async function connect() {
  const url = `postgres://${dbConfig.user}:${dbConfig.password}@${dbConfig.host}:${dbConfig.port}/${dbConfig.database}`;
  return fetch(url);
}
